### PR TITLE
SLCORE-1029: Shade/relocate JGit into OSGi bundle

### PR DESCRIPTION
**Pull request by thahnen**
[SLCORE-1029](https://sonarsource.atlassian.net/browse/SLCORE-1029)

In order for SonarLint for Eclipse to make use of the same JGit version as SonarLint CORE uses, we shade/relocate it into the OSGi bundle as well.

This PR is the adjusted cherry-pick from #1151.

[SLCORE-1029]: https://sonarsource.atlassian.net/browse/SLCORE-1029?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/java-client-osgi/pom.xml
Comment (kirill-knize-sonarsource) [inline client/java-client-osgi/pom.xml:170]: The `plugins_min_versions.txt` file will be removed in this PR:
https://github.com/SonarSource/sonarlint-core/pull/1150
It will be merged soon.

Can you please wait until it is merged and remove this line from your PR?

**Review by kirill-knize-sonarsource — CHANGES_REQUESTED** [review]

File: client/java-client-osgi/pom.xml
Comment (thahnen) [reply client/java-client-osgi/pom.xml:170]: Sure, I will rebase then.

Comment (damien-urruty-sonarsource) [reply client/java-client-osgi/pom.xml:170]: Merged

**Review by kirill-knize-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**
